src/libs/actions/IOU.js

```diff
@@ -535,41 +540,103 @@ function splitBillAndOpenReport(participants, currentUserLogin, amount, comment,
+        amount,
+        moneyRequestAction.originalMessage.currency,
+        moneyRequestAction.originalMessage.comment,
+        [],
+        '',
         transactionID,
-        comment,
-    })
-        .then((response) => {
-            if (response.jsonCode !== 200) {
-                Log.hmmm('Error rejecting transaction', {error: response.error});
-                return;
-            }
+        iouReportID,
+    );
 
-            const chatReport = response.reports[chatReportID];
-            const iouReport = response.reports[reportID];
-            Report.syncChatAndIOUReports(chatReport, iouReport);
-        })
-        .finally(() => {
-            // Setting as null deletes the transactionID
-            Onyx.merge(ONYXKEYS.TRANSACTIONS_BEING_REJECTED, {
-                [transactionID]: null,
-            });
-        });
+    const currentUserEmail = optimisticReportAction.actorEmail;
+    const updatedIOUReport = IOUUtils.updateIOUOwnerAndTotal(iouReport, currentUserEmail, amount, type);
+
+    chatReport.maxSequenceNumber = newSequenceNumber;
+    chatReport.lastReadSequenceNumber = newSequenceNumber;
+    chatReport.lastMessageText = optimisticReportAction.message[0].text;
+    chatReport.lastMessageHtml = optimisticReportAction.message[0].html;
+    chatReport.hasOutstandingIOU = updatedIOUReport.total !== 0;
+
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReportID}`,
+            value: {
+                [optimisticReportAction.sequenceNumber]: {
+                    ...optimisticReportAction,
+                    pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                },
+            },
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${chatReportID}`,
+            value: chatReport,
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_IOUS}${iouReportID}`,
+            value: updatedIOUReport,
+        },
+    ];
+    const successData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReportID}`,
+            value: {
+                [optimisticReportAction.sequenceNumber]: {
+                    pendingAction: null,
+                },
+            },
+        },
+    ];
+    const failureData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReportID}`,
+            value: {
+                [optimisticReportAction.sequenceNumber]: {
+                    ...optimisticReportAction,
+                    pendingAction: null,
+                    errors: {
+                        [DateUtils.getMicroseconds()]: Localize.translateLocal('iou.error.genericCreateFailureMessage'),
+                    },
```

Comment:
Need update error message, we are canceling here.

Reply:
Updated!